### ISSUE TITLE
Beta history: UI less informative than old history

### ISSUE THREAD

**Issue by jdavcs** [galaxyproject/galaxy#12493]
**Describe the bug**
The beta history provides less immediately-visible information about the history items:
- does not display number of items shown/hidden/etc.
- item text is cut-off, resulting in incomplete item names and not displayed number of collection items

This makes the beta history's UI not as useful as the current one. 

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 21.09.dev0 
Commit: 3c3d9e493cbaf69473e427d13ef1054076c680a3
OS: Linux
browser: Chrome 93.0.4577.63

**To Reproduce**
1. Enable beta history
2. Load some datasets, make a collection; hide some items.
3. Compare what information about the history and the datasets you see in the beta history UI to the old history UI.

**Expected behavior**
The old history is more informative.

**Screenshots**
![screen1](https://user-images.githubusercontent.com/6132258/133824168-aed07699-bf4f-4aa5-bd44-9f5531773912.png)

**Comment by dannon** [comment]
Adding the counts at the top should be a straightforward addition for a new contributor -- we should already have the data locally and simply need to render it.

**Comment by bgruening** [comment]
UX question, what does the count indicate? Is there more use than an indication of how many datasets are hidden and deleted? The latter should maybe go into the new selection, grey bar? Maybe on mouseover?

Do we need this information to be seen all the time and waste space?

**Comment by davelopez** [comment]
IMHO I think the datasets counts should be pretty clear to the user at all times, otherwise, it may cause confusion when you have a lot of hidden datasets or you have an *empty* history with all datasets deleted (but not purged) or you go back to a really old history that you don't remember that well what was in... and situations like that.

I will also be in favor of counting the number of datasets matching the current filter (which we have now) so it's more obvious that a filter is in place and you are seeing a subset of the history.

If the space is so important the information can be shown on hover of course but there may be users that don't hover around stuff so frequently and be unaware of that functionality. Maybe offer the option to *collapse* the information but be visible by default?

**Comment by guerler** [comment]
I don't think its a space issue, the only reason I avoided displaying that information was because the data was not in sync due to the multiple store issue and that we started of watching history items not the history object itself. But since #13721 this is not an issue anymore, so I'd suggest to just restore the display similar to the previous history. The data is now already available in the component and in sync.
